### PR TITLE
[UI] Improve layout of worker pool view

### LINKED ISSUE #1413: [UI] Improve layout of worker pool view
**Reserved** for [outreachy](https://www.outreachy.org/) candidates. In order for this to be assigned to you, make sure to complete the first step of the application process outlined in the [project description](https://www.outreachy.org/december-2019-to-march-2020-internship-round/communities/mozilla/#build-a-component-that-can-display-a-json-schema).

---

**Describe the bug**
In an effort to [improve the layout of our forms](https://github.com/taskcluster/taskcluster/issues/835), let's have the worker pool take in some of the [changes that were recently applied to the hook view](https://github.com/taskcluster/taskcluster/pull/1410#issue-322846137). Some of the changes that could be beneficial to this view:

* Reduce the spacing between the owner's email and the switch

Screen Shot 2019-09-30 at 1 40 32 PM

* Make spacing between each section similar to the one in the hooks view
* Make spacing between the section label and its content similar to the hook view. For example, we can reduce the following spacing between "Configuration" and the editor:

Screen Shot 2019-09-30 at 1 43 28 PM

**To Reproduce**
1. Go to https://taskcluster-ui.herokuapp.com/worker-manager/taskcluster-imaging%2Ftaskcluster-imaging-ci

That's it :)

**Expected behavior**
The overall layout is consistent with the hook view (e.g., https://taskcluster-ui.herokuapp.com/hooks/garbage/abc456)

### PR DESCRIPTION
**Pull request by Henikilan**
Resolves #1413.

**Before**
before

**After**
after

### CODE REVIEW

File: ui/src/components/WMWorkerPoolEditor/index.jsx [hunk 1]
```diff
@@ -354,104 +346,110 @@ export default class WMWorkerPoolEditor extends Component {
...
-          <ListSubheader>Configuration</ListSubheader>
-          <ListItem>
-            <CodeEditor
-              value={JSON.stringify(workerPool.config, null, 2)}
-              onChange={this.handleEditorChange}
-              lint
+            <ListItemText
+              disableTypography
+              primary={
+                <Typography variant="subtitle1">Configuration</Typography>
```
Comment (helfi92) [inline ui/src/components/WMWorkerPoolEditor/index.jsx]: Let's add a `gutterBottom` prop for some more spacing in the bottom.

```diff
diff --git a/ui/src/components/WMWorkerPoolEditor/index.jsx b/ui/src/components/WMWorkerPoolEditor/index.jsx
index dbea798ed..b42c9312f 100644
--- a/ui/src/components/WMWorkerPoolEditor/index.jsx
+++ b/ui/src/components/WMWorkerPoolEditor/index.jsx
@@ -386,7 +386,9 @@ export default class WMWorkerPoolEditor extends Component {
             <ListItemText
               disableTypography
               primary={
-                <Typography variant="subtitle1">Configuration</Typography>
+                <Typography gutterBottom variant="subtitle1">
+                  Configuration
+                </Typography>
               }
               secondary={
                 <CodeEditor
```

File: ui/src/components/WMWorkerPoolEditor/index.jsx [hunk 2]
```diff
@@ -300,37 +294,36 @@ export default class WMWorkerPoolEditor extends Component {
...
-          </ListItem>
-        </List>
-
-        <List className={classes.list}>
+        <List>
+          <div className={classes.workerPoolIdDiv}>
+            <ListItem>
+              <TextField
+                name="workerPoolId1"
+                label="Worker Pool ID 1"
```
Comment (helfi92) [inline ui/src/components/WMWorkerPoolEditor/index.jsx]: I think we can keep the old look for the worker pool. Having to write "1" and "2" is not the best :)

Screen Shot 2019-10-10 at 2 43 23 PM

File: ui/src/components/WMWorkerPoolEditor/index.jsx [hunk 3]
```diff
@@ -354,104 +346,110 @@ export default class WMWorkerPoolEditor extends Component {
...
+                control={
+                  <Switch
+                    checked={workerPool.emailOnError}
+                    onChange={this.handleSwitchChange}
+                    value="emailOnError"
+                  />
+                }
+                label="Email the owner about errors"
+              />
+            </FormGroup>
```
Comment (helfi92) [inline ui/src/components/WMWorkerPoolEditor/index.jsx]: 👍

**Comment by Henikilan** [comment]
@helfi92 
Thank you, I have just updated PR with fixes

**Comment by helfi92** [comment]
@all-contributors please add @Henikilan for code.

**Pull request merged**